src/components/Interpretations/InterpretationsProvider/InterpretationsManager.js

```diff
@@ -0,0 +1,307 @@
+        this.interpretations.clear()
+    }
+
+    resetInterpretations(newInterpretations) {
+        this.interpretations.clear()
+        for (const interpretation of newInterpretations) {
+            this.interpretations.set(interpretation.id, interpretation)
+        }
+    }
+
+    async fetchInterpretationDetails(id) {
+        const result = await this.query({
+            interpretation: {
+                resource: 'interpretations',
+                id,
+                params: {
+                    fields: [
+                        'access[write,manage]',
+                        'comments[id,text,created,createdBy[id,displayName]]',
+                        'created',
+                        'createdBy[id,displayName]',
+                        'id',
+                        'likedBy',
+                        'likes',
+                        'text',
+                    ],
+                },
+            },
+        })
+        return result.interpretation
+    }
+
+    async fetchInterpretationsList() {
+        if (!this.currentType || !this.currentVisualizationId) {
+            throw new Error(
+                'Called fetchInterpretationsList before currentType or currentVisualizationId was set'
+            )
+        }
+        const result = await this.query({
+            interpretations: {
+                resource: 'interpretations',
+                params: {
+                    fields: [
+                        'access[write,manage]',
+                        'comments[id]',
+                        'created',
+                        'createdBy[id,displayName]',
+                        'id',
+                        'likedBy[id]',
+                        'likes',
+                        'text',
+                    ],
+                    filter: `${this.currentType}.id:eq:${this.currentVisualizationId}`,
+                    paging: false,
+                },
+            },
+        })
+        return result.interpretations.interpretations
+    }
+
+    async loadActiveInterpretation(id) {
+        const interpretation = await this.fetchInterpretationDetails(id)
+        this.interpretations.set(id, interpretation)
+        this.activeInterpretationId = id
+        return this.getActiveInterpretation()
+    }
+
+    async loadInterpretationsForVisualization(type, id) {
+        this.currentType = type
+        this.currentVisualizationId = id
+        const interpretations = await this.fetchInterpretationsList()
+        this.resetInterpretations(interpretations)
+        return groupInterpretationIdsByDate(interpretations)
+    }
+
+    async createInterpretation({ type, id, text, onComplete }) {
+        await this.mutate(
+            {
+                resource: `interpretations/${type}/${id}`,
+                type: 'create',
+                data: text,
+            },
+            { onComplete }
+        )
+        /* since the create request does not rerun an ID we must refetch the list
```

Comment:
```suggestion
        /* since the create request does not return an ID we must refetch the list
```
I guess you meant `return` here.